packages/navigator/src/browser/navigator-editor-sync.ts

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (C) 2018 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable, inject, postConstruct } from "inversify";
+import { Disposable } from '@theia/core/lib/common';
+import { EditorManager, EditorWidget } from "@theia/editor/lib/browser";
+import { ITreeNode, ISelectableTreeNode, IExpandableTreeNode, PreferenceChangeEvent } from "@theia/core/lib/browser";
+import { FileNavigatorModel } from "./navigator-model";
+import { NavigatorConfiguration, NavigatorPreferences } from "./navigator-preferences";
+
+@injectable()
+export class NavigatorEditorSynchronizer {
+
+    @inject(FileNavigatorModel)
+    private readonly fileNavigatorModel: FileNavigatorModel;
+
+    @inject(EditorManager)
+    private readonly editorManager: EditorManager;
```

Comment:
Instead of watching the editorManager, we should watch the shell, for active widgets that 'have a URI'. Then it will work for preview widgets and other kinds of editors and doesn't introduce a hard dependency to the editor extension.

Reply:
I did something like [this](https://github.com/theia-ide/theia/pull/1383/files#diff-f9afc1d61ee7ba79952f1f81bdf6bd52R95).

Reply:
Thanks, will take a look